src/Select.js

```diff
@@ -339,31 +340,38 @@ const Select = React.createClass({
 			});
 		}
 
+		const inputLabel = this.getInputLabel();
+
+		// determines if the user mouse downed on the actual ValueComponent excluding placeholder text
+		const isOnValue = this.valueComponent && (this.valueComponent.valueNode === event.target) ? true : false;
+
+		// clears value so that the cursor will be a the end of input then the component re-renders
+		this.input.getInput().value = '';
```

Comment:
Just like the check below we should probably check `getInput` is a function before we access it.